core/templates/pages/exploration-editor-page/feedback-tab/feedback-tab.component.spec.ts

```diff
@@ -0,0 +1,415 @@
+        last_updated_msecs: 0,
+        message_count: 1,
+        thread_id: '1',
+        last_nonempty_message_author: 'Message 1',
+        last_nonempty_message_text: 'Message 2'
+      }, {
+        suggestion_type: 'edit_exploration_state_content',
+        suggestion_id: '1',
+        target_type: '',
+        target_id: '',
+        status: '',
+        author_name: '',
+        change: {
+          state_name: '',
+          new_value: '',
+          old_value: '',
+        },
+        last_updated_msecs: 0
+      }));
+    spyOn(threadDataService, 'getMessagesAsync').and.returnValue(
+      $q.resolve());
+    ctrl.setActiveThread('1');
+    $scope.$apply();
+
+    spyOn(suggestionModalForExplorationEditorService, 'showSuggestionModal')
+      .and.callFake(function(suggestionType, obj) {
+        obj.setActiveThread('1');
+      });
+    ctrl.showSuggestionModal();
+    $scope.$apply();
+  });
+
+  it('should create a new thread when closing create new thread modal',
+    function() {
+      spyOn(alertsService, 'addSuccessMessage').and.callThrough();
+      spyOn(threadDataService, 'createNewThreadAsync').and.returnValue(
+        $q.resolve());
+      spyOn($uibModal, 'open').and.returnValue({
+        result: $q.resolve({
+          newThreadSubject: 'New subject',
+          newThreadText: 'New text'
+        })
+      });
+      ctrl.showCreateThreadModal();
+      $scope.$apply();
+      $scope.$apply();
+
+      expect(threadDataService.createNewThreadAsync)
+        .toHaveBeenCalledWith('New subject', 'New text');
+      expect(alertsService.addSuccessMessage).toHaveBeenCalledWith(
+        'Feedback thread created.');
+      expect(ctrl.tmpMessage.status).toBe(null);
+      expect(ctrl.tmpMessage.text).toBe('');
+    });
+
+  it('should not create a new thread when dismissing create new thread modal',
+    function() {
+      spyOn(threadDataService, 'createNewThreadAsync');
+      spyOn($uibModal, 'open').and.returnValue({
+        result: $q.reject()
+      });
+      ctrl.showCreateThreadModal();
+      $scope.$apply();
+
+      expect(threadDataService.createNewThreadAsync).not.toHaveBeenCalled();
+    });
+
+  it('should get css classes based on status', function() {
+    expect(ctrl.getLabelClass('open')).toBe('badge badge-info');
+    expect(ctrl.getLabelClass('compliment')).toBe('badge badge-success');
+    expect(ctrl.getLabelClass('another')).toBe('badge badge-secondary');
+  });
+
+  it('should get human readable status from provided status', function() {
+    expect(ctrl.getHumanReadableStatus('open')).toBe('Open');
+    expect(ctrl.getHumanReadableStatus('compliment')).toBe('Compliment');
+    expect(ctrl.getHumanReadableStatus('not_actionable')).toBe(
+      'Not Actionable');
+  });
+
+  it('shoud get locale date time string', function() {
+    // This corresponds to Fri, 21 Nov 2014 09:45:00 GMT.
+    var NOW_MILLIS = 1416563100000;
+    expect(ctrl.getLocaleAbbreviatedDatetimeString(NOW_MILLIS)).toBe(
```

Comment:
Any timezone related problems with this call? I think we should mock it right?

Reply:
Done.